### PR TITLE
Fix: hparams.yaml saved twice when using TensorBoardLogger

### LINKED ISSUE #5562: hparams.yaml saved twice when using TensorBoardLogger
## 🐛 Bug

When using `TensorBoardLogger(...default_hp_metric=False...)` and calling `self.save_hyperparameters()` from `LightningModule `instance, two _hparams.yaml_ files are created (unexpected one in the logger's `save_dir` and second under proper experiment version dir).

## Please reproduce using the BoringModel
See [**BoringModel**](https://colab.research.google.com/drive/1OpUpfU9awCEjPCR_XYo8z5TmF0MxBTJA?usp=sharing).

### To Reproduce

1. Create `LightningModule` with `self.save_hyperparameters()` call in `__init__`
2. Create `TensorBoardLogger` with `default_hp_metric=False`
3. Start training

### Expected behavior

Only one _hparams.yaml_ in `save_dir/name/version` should be created (like it happens when using tb logger with `default_hp_metric=True`).

### Environment
From Collab Notebook:

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.1.4
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           1 SMP Thu Jul 23 08:00:38 PDT 2020

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #5562 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by carmocca** [comment]
Does the test fail in current master?

**Comment by kaushikb11** [comment]
@carmocca Good Q. Updated!

File: tests/loggers/test_tensorboard.py [hunk 1]
```diff
@@ -291,3 +292,19 @@ def test_tensorboard_finalize(summary_writer, tmpdir):
...
+    logger = TensorBoardLogger(save_dir=tmpdir, default_hp_metric=False)
+    trainer = Trainer(max_steps=1, default_root_dir=tmpdir, logger=logger)
+    assert trainer.log_dir == trainer.logger.log_dir
+    trainer.fit(model)
+
+    dir_path = trainer.log_dir
+
+    hparams_file_path = os.path.join(dir_path, hparams_file)
+    assert os.path.isfile(hparams_file_path)
+    assert not os.path.isfile(Path(tmpdir / hparams_file))
```
Comment (carmocca) [inline tests/loggers/test_tensorboard.py:310]: Minor cleanup:

```suggestion
    model = BoringModel()
    logger = TensorBoardLogger(save_dir=tmpdir, default_hp_metric=False)
    trainer = Trainer(max_steps=1, default_root_dir=tmpdir, logger=logger)
    assert trainer.log_dir == trainer.logger.log_dir
    trainer.fit(model)

    hparams_file = "hparams.yaml"
    assert os.path.isfile(os.path.join(trainer.log_dir, hparams_file))
    assert not os.path.isfile(os.path.join(tmpdir, hparams_file))
```

File: tests/loggers/test_tensorboard.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 import os
 from argparse import Namespace
 from distutils.version import LooseVersion
+from pathlib import Path
```
Comment (carmocca) [inline tests/loggers/test_tensorboard.py:17]:
```suggestion
```

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by SkafteNicki — APPROVED** [review]
